src/main/java/dk/cngroup/trainings/spring/springassignment/controller/AnimalController.java

```diff
@@ -14,61 +15,109 @@
+												   @RequestBody @Valid Animal animal) {
+		if (animalService.getAnimalById(id).isPresent()) {
+			animal.setId(id);
+			return this.addAnimal(animal);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
 
-    @RequestMapping(path = "/{id}", method = RequestMethod.GET)
-    public ResponseEntity<Animal> getAnimalById(@PathVariable("id") long id){
-        Optional<Animal> searchedAnimal= animalService.getAnimalById(id);
-        if(searchedAnimal!=null)
-            return new ResponseEntity<>(searchedAnimal.get(),HttpStatus.OK);
-        else
-            return ResponseEntity.notFound().build();
-    }
+	@RequestMapping(path = "/{id}", method = RequestMethod.GET)
+	public ResponseEntity<Animal> getAnimalById(@PathVariable("id") long id) {
+		Optional<Animal> searchedAnimal = animalService.getAnimalById(id);
+		if (searchedAnimal.isPresent()) {
+			return new ResponseEntity<>(searchedAnimal.get(), HttpStatus.OK);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
 
-    @RequestMapping(path = "/search/{name}", method = RequestMethod.GET)
-    public ResponseEntity<List<Animal>> getAnimalByName(@PathVariable("name") String name){
-        List<Animal> sameNameAnimals= animalService.getAnimalsByName(name);
-        if(sameNameAnimals.size()!=0)
-            return new ResponseEntity<>(sameNameAnimals, HttpStatus.OK);
-        else
-            return ResponseEntity.notFound().build();
-    }
+	@RequestMapping(path = "/name", method = RequestMethod.GET)
+	public ResponseEntity<List<Animal>> getAnimalsByName(@RequestParam("name") String name) {
+		List<Animal> sameNameAnimals = animalService.getAnimalsByName(name);
+		if (!sameNameAnimals.isEmpty()) {
+			return new ResponseEntity<>(sameNameAnimals, HttpStatus.OK);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	@RequestMapping(path = "/{id}/careTakers", method = RequestMethod.POST)
+	public ResponseEntity<CareTaker> addNewCareTakerToExistingAnimal(@PathVariable("id") long id,
+																	 @RequestBody @Valid CareTaker careTaker) {
+		if (animalService.getAnimalById(id).isPresent()) {
+			Optional<CareTaker> addedCareTaker = animalService.addNewCareTakerToExistingAnimal(id, careTaker);
+			if (addedCareTaker.isPresent()) {
+				return new ResponseEntity<>(addedCareTaker.get(), HttpStatus.OK);
+			} else {
+				return ResponseEntity.badRequest().build();
+			}
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	@RequestMapping(value = "/{id}/careTakers", method = RequestMethod.GET)
+	public ResponseEntity<List<CareTaker>> getCareTakersByAnimalId(@PathVariable("id") long id) {
+		Optional<Animal> animalById = animalService.getAnimalById(id);
+		if (animalById.isPresent()) {
+			return new ResponseEntity<>(animalById.get().getCareTakers()
+					, HttpStatus.OK);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	/*
+	Is it Ok if I injected CareTakerController and I used its method
```

Comment:
well you are not injecting the Controller here, are you?

Reply:
No I didn't, because I remembered you saying that controllers should not depend on each other.